fades/helpers.py

```diff
@@ -208,3 +210,37 @@ def check_pypi_updates(dependencies):
 
     dependencies["pypi"] = dependencies_up_to_date
     return dependencies
+
+
+def _pypi_head_package(pkg_name):
+    """Hit pypi with a http HEAD to check if pkg_name exists."""
+    url = BASE_PYPI_URL.format(name=pkg_name)
+    req = request.Request(url, method='HEAD')
+    try:
+        response = request.urlopen(req)
+    except HTTPError as http_error:
+        if http_error.code == HTTP_STATUS_NOT_FOUND:
+            return False
+        else:
+            raise http_error
+    else:
+        if response.status == HTTP_STATUS_OK:
+            logger.debug("%s exists in Pypi.", pkg_name)
+            return True
+
+
+def check_pypi_exists(dependencies):
+    """Check if the indicated dependencies actually exists in pypi."""
+    for dependency in dependencies.get('pypi', []):
+        logger.debug("Checking if %s exists in PyPi", dependency)
```

Comment:
I'd use %r, just in case, because we may be hitting the issue of "hey, it fails to install dependency foobar why?" and the problem being that somehow the dependency verified is "foobar ", not "foobar"

Reply:
done